packages/lib/src/artifacts/Contract.ts

```diff
@@ -82,8 +82,9 @@ function _wrapContractInstance(schema: any, instance: Web3Contract): Contract {
 
   instance.at = function(address: string): Contract | never {
     if(!ZWeb3.isAddress(address)) throw new Error('Given address is not valid: ' + address);
-    instance.options.address = instance._address = address;
-    return instance;
+    const newWeb3Instance = instance.clone();
+    newWeb3Instance.options.address = newWeb3Instance._address = address;
```

Comment:
Can we not use chained assignments? I believe they are error prone because how they are implemented in JS and just in general not easy flow. Still a bit opinionated statement. Would be great to have shared guidelines.